Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -1361,6 +1364,50 @@ extension Tensor {
         "Axes must have rank 0 or rank 1; axes has rank \(axes.rank) with values \(axes.scalars).",
         file: file,
         line: line)
-    return areAxesInRange(axes.scalars)
+    return isInRange(axes: axes.scalars)
+  }
+
+  /// Ensures the provided axes refer to axis of `self` (are in range `-rank..<rank`).
+  @usableFromInline
+  func ensureValid(
+      axes: Tensor<Int32>,
+      function: StaticString = #function,
+      file: StaticString = #file,
+      line: UInt = #line
+  ) {
+      precondition(
+          isInRange(axes: axes, file: file, line: line),
+          "All axes must be in `-rank..<rank` when calling \(function) (rank: \(rank), axes: \(axes))",
```

Comment:
C.f. https://github.com/tensorflow/swift-apis/pull/673#discussion_r394652982, What purpose do the backticks serve here?  If we want to set a precedent that all code embedded in error messages should be markdown-ified, I guess that's fine with me (and we should write it down somewhere), but there's no precedent for it.  I note you don't have backticks around the function name.

Reply:
I thought that it was more readable when it is clear that we've switched from English text to Swift-equivalent code, as the syntax `-rank..<rank` is not normal written English, whereas everything else (including the function name) is reasonable English. Does that make sense? Do you think it reads better with the backticks removed?

Reply:
Not necessarily; it's just that there's no precedent for using backticks this way when printing to the console AFAIK.  The swift compiler, for example, encloses type names in single quotes when printing diagnostics.  Backticks can work too I suppose, but we should choose consciously and record what we choose.  One argument in favor of backticks is that I'd guess people who are used to writing markdown may gravitate toward them naturally.  You might even claim that Markdown is now lingua franca for text with embedded code, which makes backticks “the right” answer.